File: backend/src/main/java/com/ufc/backend/backend/resources/FeedbackResource.java
package com.ufc.backend.backend.resources;

import com.ufc.backend.backend.model.feedback.Feedback;
import com.ufc.backend.backend.services.FeedbackService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.support.ServletUriComponentsBuilder;

@RestController
@RequestMapping(value = "/feedback")
@CrossOrigin
public class FeedbackResource {

    @Autowired
    private FeedbackService service;

    @PostMapping
    public ResponseEntity<Feedback> insert(@RequestBody Feedback feedback) {
        return ResponseEntity.created(ServletUriComponentsBuilder
                .fromCurrentRequest()
                .path("/feedback")
                .buildAndExpand(feedback.getId())
                .toUri()).body(service.save(feedback));
    }

    @PutMapping
    public ResponseEntity<Feedback> update(@RequestBody Feedback feedback) {
        return ResponseEntity.ok().body(service.update(feedback));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable String id) {
        service.delete(id);
        return ResponseEntity.ok().build();
    }



}

File: backend/src/main/java/com/ufc/backend/backend/resources/FeedbackResource.kt
package com.ufc.backend.backend.resources

import com.ufc.backend.backend.model.feedback.Feedback
import com.ufc.backend.backend.services.FeedbackService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.servlet.support.ServletUriComponentsBuilder

@RestController
@RequestMapping(value = ["/feedback"])
@CrossOrigin
class FeedbackResource(
    private val service: FeedbackService
) {
    @PostMapping("/{id}")
    fun insert(@RequestBody feedback: Feedback, @PathVariable id: String): ResponseEntity<Feedback> {
        return ResponseEntity.created(
            ServletUriComponentsBuilder
                .fromCurrentRequest()
                .path("/feedback")
                .buildAndExpand(feedback.id)
                .toUri()
        ).body(service.save(feedback, id))
    }

    @PutMapping("/{id}")
    fun update(@RequestBody feedback: Feedback, @PathVariable id: String): ResponseEntity<Feedback> {
        return ResponseEntity.ok().body(service.save(feedback, id))
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: String): ResponseEntity<Void> {
        service.delete(id)
        return ResponseEntity.ok().build()
    }
}